### DISCUSSION TITLE
get_descriptor() and equivalent control requests don't cause any USB bus activity

### DISCUSSION THREAD

**Discussion by BobSki778** [pyusb/pyusb#426] [Q&A] [answered]
Hi,

For reasons, I need to cause actual activity on the USB hardware lines (and communication to a USB device), such as requesting the device descriptor.  However, then I run these commands, pyUSB seems to be returning a "cached" copy of the descriptor, rather than actually requesting it from the device.  I am using a USB Bus protocol analyzer to monitor activity on the physical USB bus.  I can see initial enumaration when I attach the device, but "get" pyUSB commands cause no bus activity, while "set" pyUSB commands (such as dev.set_configuration(0)) do.

I am on macOS Monterey 12.4, and have installed libusb 1.0.26 via homebrew

This is with Python 3.7.3.

Also, for what it's worth, pyUSB doesn't seem to be able to find the backend unless I specify it explicitly.  That's probably a path issue, and I'm less concerned with solving that at the moment.

Example code:

```
import usb.core
import usb.control
import usb.backend.libusb1 as libusb1

backend = libusb1.get_backend(find_library=lambda x: "/opt/brew/Cellar/libusb/1.0.26/lib/libusb-1.0.dylib")
dev = usb.core.find(idVendor=MyVendorID, idProduct=MyProductID)

usb.control.get_descriptor(dev, 18, 1, 0) # get device descriptor from pyUSB usb.control module: causes no bus activity, but does return expected data

# equivalent request using lower level usb.core and usb.util functions: causes no bus activity, but does return expected data
import usb.util
bmRequestType = usb.util.build_request_type(usb.util.CTRL_IN, usb.util.CTRL_TYPE_STANDARD, usb.util.CTRL_RECIPIENT_DEVICE)
dev.ctrl_transfer(bmRequestType=bmRequestType, bRequest=0x06, wValue=0x0100, wIndex=0, data_or_wLength=18)

# these cause actual bus activity, but only if the requested configuration value differs from the present configuration value
dev.set_configuration(0)
dev.set_configuration(1)
```

**Comment by mcuee** [comment, marked as answer]
You may have to issue explicit control request to trigger bus activity.

**Comment by mcuee** [comment]
For the other question, please refer to the FAQ here.
https://github.com/pyusb/pyusb/blob/master/docs/faq.rst#how-do-i-fix-no-backend-available-errors
